lib/nbp/push_connector.rb

```diff
@@ -0,0 +1,152 @@
+# frozen_string_literal: true
+
+require 'singleton'
+require 'concurrent'
+
+module Nbp
+  class PushConnector
+    include Singleton
+
+    class Error < StandardError; end
+    class SettingsError < Error; end
+    class ServerError < Error; end
+
+    def initialize
+      super
+      @token = Concurrent::ThreadLocalVar.new
+      @token_expiration = Concurrent::ThreadLocalVar.new
+
+      create_source! unless source_exists?
+    end
+
+    def self.enabled?
+      @enabled ||= Settings.nbp&.push_connector&.enable || false
+    end
+
+    def push_lom!(xml)
+      response = api_conn.put("/push-connector/api/lom-v2/#{source_slug}") do |req|
+        req.body = {metadata: xml}.to_json
+      end
+      raise ServerError if response.status == 500
+
+      raise_connector_error('Could not push task LOM', response) unless response.success?
+    end
+
+    def delete_task!(task_uuid)
+      response = api_conn.delete("/push-connector/api/course/#{source_slug}/#{task_uuid}")
+      raise ServerError if response.status == 500
+
+      raise_connector_error('Could not delete task', response) unless response.success? || response.status == 404
+    end
+
+    def process_uploaded_task_uuids(&)
+      offset = 0
+      until (uuids = get_uploaded_task_uuids(offset)).empty?
+        offset += uuids.length
+        uuids.map(&)
+      end
+    end
+
+    def source_exists?
+      response = api_conn.get("/datenraum/api/core/sources/slug/#{source_slug}")
+      if response.status == 200
+        true
+      elsif response.status == 404
+        false
+      else
+        raise_connector_error('Could not determine if source exists', response)
+      end
+    end
+
+    private
+
+    # the NBP API does not accept limits over 100
```

Comment:
Wouldn't this be a dedicated check, so that we `raise` in case a limit > 100 is requested?